ClickHouse.Client.Tests/SqlSelectTests.cs

```diff
@@ -204,21 +204,53 @@ public async Task ShouldGetReaderSchemaTable()
             var schema = reader.GetSchemaTable();
             Assert.AreEqual(2, schema.Rows.Count);
         }
+        
+        public static IEnumerable<TestCaseData> ParametersQueries => TestUtilities.GetDataTypeSamples()
+            //.Where(sample => !new [] {"Enum", "DateTime64(9)"}.Contains(sample.ClickHouseType)) //old clh doesn`t know about regular Enum and DateTime64
+            .Where(sample => sample.ExampleValue != DBNull.Value) //null value should be handled by writing "is null" statement
+            .Where(sample => sample.ClickHouseType != "Date") //DateTime with 00:00:00 can`t be Date type, cause it`ll broke DateTime logic. Specify type to fix that
+            .Where(sample => sample.ClickHouseType != "DateTime64(9)") //Default DateTime CLH analog is DateTime. If you want to use DateTime64 specify type as parameter
+            .Where(sample => sample.ClickHouseType != "UUID") // https://github.com/ClickHouse/ClickHouse/issues/7463
+            .Select(sample => new TestCaseData(
+                $"SELECT * FROM (SELECT {sample.ExampleExpression} AS res) WHERE {sample.WhereClause}",
```

Comment:
Testing 2 things at once, not a good idea (hard to understand what to fix if something breaks). `SELECT` and `WHERE` should be tested separately

Reply:
done